test/gilded-rose.spec.ts

```diff
@@ -4,6 +4,275 @@ describe('Gilded Rose', () => {
   it('should foo', () => {
     const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
     const items = gildedRose.updateQuality();
-    expect(items[0].name).toBe('fixme');
+    expect(items[0].name).toBe('foo');
   });
+
+  it('given foo is in inventory, when 10 days passes, then sellIn decreases by 10', () => {
+    const gildedRose = new GildedRose([new Item('foo', 15, 0)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].sellIn).toBe(5);
+  })
+
+  it('given foo is in inventory, when 10 days passes, then quality decreases by 10', () => {
+    const gildedRose = new GildedRose([new Item('foo', 15, 15)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(5);
+  })
+
+  it('given foo is in inventory and sellIn is negative, when 10 days passes, then quality decreases by 20', () => {
+    const gildedRose = new GildedRose([new Item('foo', -5, 30)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(10);
+  })
+
+  it('given foo is in inventory and sellIn is 5, when 10 days passes, then quality decreases by 15', () => {
+    const gildedRose = new GildedRose([new Item('foo', 5, 30)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(15);
+  })
+
+  it('given foo is in inventory and quality is 5, when 10 days passes, then quality is 0', () => {
+    const gildedRose = new GildedRose([new Item('foo', 15, 5)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(0);
+  })
+
+  it('given Aged Brie is in inventory, when 10 days passes, then quality increase by 10', () => {
+    const gildedRose = new GildedRose([new Item('Aged Brie', 20, 10)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
+
+    expect(items[0].quality).toBe(20);
+  })
+
+  it('given Aged Brie is in inventory and sellIn is negative, when 10 days passes, then quality increase by 20', () => {
+    const gildedRose = new GildedRose([new Item('Aged Brie', -10, 10)]);
+
+    let items;
+    for (let i = 0; i < 10; i++) items = gildedRose.updateQuality();
```

Comment:
My personal preference is to never use single-line blocks, just a FYI, don't bother changing this